app/controllers/api/notifications_controller.rb

```diff
@@ -1,5 +1,30 @@
 module Api
+  # NOTE: This uses the NotificationRecipients model to then map to notifications
   class NotificationsController < BaseController
+    # based upon BaseController#index
+    def index
+      klass = collection_class(@req.subject)
+      res, subquery_count = collection_search(@req.subcollection?, @req.subject, klass)
+
+      res_count = (res.kind_of?(ActiveRecord::Relation) ? res.except(:select) : res).count
```

Comment:
@Fryguy maybe we need to fix the counts so they can be used in the ui

Reply:
I'm not sure what you mean - can you elaborate how this is broken?

Reply:
Is this the fact that subcount is incorrect?  If so, yes, I agree, but maybe not here? I'm also wondering if subcount is broken in general.

Reply:
I've reviewed all this and subcount and count are both correct.  I don't know what problem the UI is trying to solve, but yes we need to do it in the first query as part of the backend API somehow.  There's a third value called subquery_count, which I think comes from when you use a filter, but it's really hard to tell, and that looks buggy too, but perhaps that's what's needed here?

Reply:
ugh, sorry I put this comment in here. This thread belongs in https://github.com/ManageIQ/manageiq-ui-classic/issues/9448

commenting over there

Reply:
So, does it make sense for us to add the count here so we fix the double query.

I had thought these 2 issues were unrelated, but maybe the solution is modifying this query.